src/WebJobs.Extensions.DurableTask/AzureStorageDurabilityProviderFactory.cs

```diff
@@ -27,12 +28,21 @@ internal class AzureStorageDurabilityProviderFactory : IDurabilityProviderFactor
             IOptions<DurableTaskOptions> options,
             IConnectionStringResolver connectionStringResolver,
             INameResolver nameResolver,
-            ILoggerFactory loggerFactory)
+            ILoggerFactory loggerFactory,
+            PlatformInformation platformInfo = null)
```

Comment:
You should likely make PlatformInformation a service, and then inject that service via dependency injection (see [this class](https://github.com/Azure/azure-functions-durable-extension/blob/e600de93b78eb45335947741232292ea60c1530e/src/WebJobs.Extensions.DurableTask/DurableTaskJobHostConfigurationExtensions.cs#L32) and its methods called by the Functions runtime to see how we add services).

That will allow any of the other services injected (such as this one), to access your service just by adding it as a parameter in the constructor. You just need to make sure there are no circular dependencies.

This gives us flexibility in how the other backends' implementations of IDurabilityProviderFactory consume platform information.

Reply:
Note, the DI patterns used in .NET Core generally requires you to define an interface and then make a concrete implementation of that interface. You then program against the interface.

Reply:
Perfect, thanks! After reading up a bit more on how we use DI in our codebase, and DI in .NET more broadly, I have a very clear idea of what you mean now. Should be an easy refactor!